sklearn/metrics/_plot/tests/test_confusion_matrix_display.py

```diff
@@ -0,0 +1,366 @@
+    """Check that we raise the proper error when validating parameters."""
+    X, y = make_classification(
+        n_samples=100, n_informative=5, n_classes=5, random_state=0
+    )
+
+    regressor = SVR().fit(X, y)
+    y_pred_regressor = regressor.predict(X)
+    y_pred_classifier = SVC().fit(X, y).predict(X)
+
+    err_msg = "ConfusionMatrixDisplay.from_estimator only supports classifiers"
+    with pytest.raises(ValueError, match=err_msg):
+        ConfusionMatrixDisplay.from_estimator(regressor, X, y)
+
+    err_msg = "Mix type of y not allowed, got types"
+    with pytest.raises(ValueError, match=err_msg):
+        # Force `y_true` to be seen as a regression problem
+        ConfusionMatrixDisplay.from_predictions(y + 0.5, y_pred_classifier)
+    with pytest.raises(ValueError, match=err_msg):
+        ConfusionMatrixDisplay.from_predictions(y, y_pred_regressor)
+
+    err_msg = "Found input variables with inconsistent numbers of samples"
+    with pytest.raises(ValueError, match=err_msg):
+        ConfusionMatrixDisplay.from_predictions(y, y_pred_classifier[::2])
+
+
+@pytest.mark.parametrize(
+    "constructor_name", ["from_estimator", "from_predictions"]
+)
+def test_confusion_matrix_display_invalid_option(pyplot, constructor_name):
+    """Check the error raise if an invalid parameter value is passed."""
+    X, y = make_classification(
+        n_samples=100, n_informative=5, n_classes=5, random_state=0
+    )
+    classifier = SVC().fit(X, y)
+    y_pred = classifier.predict(X)
+
+    constructor = getattr(ConfusionMatrixDisplay, constructor_name)
+    params = (
+        (classifier, X, y)
+        if constructor_name == "from_estimator"
+        else (y, y_pred)
+    )
+
+    extra_params = {"normalize": "invalid"}
+
+    err_msg = r"normalize must be one of \{'true', 'pred', 'all', None\}"
+    with pytest.raises(ValueError, match=err_msg):
+        constructor(*params, **extra_params)
+
+
+@pytest.mark.parametrize(
+    "constructor_name", ["from_estimator", "from_predictions"]
+)
+@pytest.mark.parametrize("with_labels", [True, False])
+@pytest.mark.parametrize("with_display_labels", [True, False])
+def test_confusion_matrix_display_custom_labels(
+    pyplot, constructor_name, with_labels, with_display_labels
+):
+    """Check the resulting plot when labels are given."""
+    n_classes = 5
+    X, y = make_classification(
+        n_samples=100, n_informative=5, n_classes=n_classes, random_state=0
+    )
+    classifier = SVC().fit(X, y)
+    y_pred = classifier.predict(X)
+
+    constructor = getattr(ConfusionMatrixDisplay, constructor_name)
+    params = (
+        (classifier, X, y)
+        if constructor_name == "from_estimator"
+        else (y, y_pred)
+    )
+
+    ax = pyplot.gca()
+    labels = [2, 1, 0, 3, 4] if with_labels else None
+    display_labels = ["b", "d", "a", "e", "f"] if with_display_labels else None
+
+    cm = confusion_matrix(y, y_pred, labels=labels)
+    disp = constructor(
```

Comment:
In this case, I think having the `if` statement would be clearer than storing `params`

```py
common_kwargs = {'ax': ax, 'display_labels': ...}
if constructor_name == "from_estimator":
	disp = ConfustionMatrixDisplay.from_estimator(classifier, X, y, **common_kwargs)
else:
	disp = ConfusionMatrixDisplay.from_predictions(X, y_pred, **common_kwargs)
```

Same for the rest of the test we use this pattern.